Fix accordion dates

petertgiles opened the pull request:
Sorry to pull something out of the ice box.  :laughing: I needed something quick and it was making me crazy to have a bug:prod in there.

And apparently French month abbreviations are not capitalized and can have four characters?

![image](https://user-images.githubusercontent.com/8978655/202762330-389b7da2-f0f9-48db-93b1-ea7fa4d1f24d.png)

![image](https://user-images.githubusercontent.com/8978655/202762376-baf58892-4ccd-4270-8069-8d8a5b88bf67.png)


Closes #4625

mnigh reviewed (commented):
This is so much better than before!

However, I did find a Government of Canada [document](https://www.noslangues-ourlanguages.gc.ca/fr/cles-de-la-redaction/mois.html) (strangely, only available in French though a similar [document](https://www.noslangues-ourlanguages.gc.ca/en/writing-tips-plus/abbreviations-days-and-months.html) is available in English) with a table for abbreviating months in both French and English.

## Screenshot
![Screen Shot 2022-11-21 at 12 14 19](https://user-images.githubusercontent.com/3046459/203119210-f091b73d-614e-41f1-8d94-33fb688cb9ed.png)

All of the French month abbreviations work except for July, which should have an extra `l` before the period, **juill.**, and currently shows **juil.** instead. Sadly, the library we are using actually uses the incorrect abbreviation for fr-CA, but I have opened an [issue](https://github.com/date-fns/date-fns/issues/3281) for it.

Also, the Government of Canada document in English uses periods after some of the months and the [`<abbr>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr) element. I know this is changing the scope, but I'm wondering if we should be using abbreviations at all to avoid having to include markup when we could just use the full months names in any language. Thoughts?

petertgiles commented:
@Jerryescandon Thoughts on using abbreviations on this component vs using full names?  If we go with abbreviations we'll need to eventually add abbr tags for a11y but, of course, it will be longer.

Jerryescandon commented:
Abbreviations do clean up this UI a lot, and are mostly secondary info. I would love it if we can work abbreviations properly. But its by no means an important ticket, if you have the time, go for it.

petertgiles commented:
Thanks @mnigh for the l18n deep-dive on months.  I think the problems are out of scope for this PR but I've added a new issue to the board:
#4860 - For aligning abbreviations

And added a note about month abbreviations to #3313.

mnigh reviewed (approved):
👍
